src/components/edit_alarm_definition.ts

```diff
@@ -72,10 +77,46 @@ export class EditAlarmDefinitionPageCtrl {
     this.newAlarmDefinition.match_by.splice(index, 1);
   }
 
+  private _suggestAlarmActions(query, callback) {
+    return Promise.resolve(
+      this.notificationMethods.map(notification => notification.name)
+    ).then(callback);
+  }
+
+  private loadNotificationMethods() {
+    return this.monascaClientSrv
+      .listNotifications()
+      .then(
+        notification_methods =>
+          (this.notificationMethods = notification_methods)
+      );
+  }
+
+  public addAlarmAction() {
+    if (!this.newAlarmDefinition.alarm_actions_by_name) {
+      this.newAlarmDefinition.alarm_actions_by_name = [];
+    }
+    this.newAlarmDefinition.alarm_actions_by_name.push("");
+  }
+
+  public removeAlarmAction(index) {
+    if (!this.newAlarmDefinition.alarm_actions_by_name) {
+      return;
+    }
+    this.newAlarmDefinition.alarm_actions_by_name.splice(index, 1);
```

Comment:
Can you guarantee that the index of an alarm definition won't change? Would it be better to remove it by name rather than by index?

Reply:
the iterator over the list garentees order, shoudn't be a problem